src/pycropml/xml2wf.py

```diff
@@ -0,0 +1,171 @@
+from openalea.core.external import *
+from openalea.core.pkgmanager import PackageManager
+from pycropml.render_python import generate_doc
+
+
+class XmlToWf(object):
+    
+    
+    def __init__(self, xmlwf,dir, pkg_name):
+        self.xmlwf = xmlwf
+        self.dir=dir
+        self.pkg_name = pkg_name
+        self.inputLinks = self.xmlwf.inputlink
+        self.outputLinks = self.xmlwf.outputlink
+        self.internalLinks = self.xmlwf.internallink
+    
+    def run(self):
+        self.doc = generate_doc(self.xmlwf)
+        self.pkg = self.retrievePackage(self.pkg_name)
+        self.compositeNodeInputs()
+        self.compositeNodeOutputs() 
+        self.wf = CompositeNode(self.inputs_wf, self.outputs_wf)
+        self.createNodes()
+        wf_factory = CompositeNodeFactory(self.xmlwf.name+"_wf", doc=self.doc)
+        self.connectOutputs()
+        self.connectInputs()
+        self.connectInternal()
+        self.wf.to_factory(wf_factory)
+        if wf_factory.name in self.pkg:
+            del self.pkg[wf_factory.name]
+        self.pkg.add_factory(wf_factory)
+
+        self.pkg.write()
+        
+    def retrievePackage(self, name):
+         pm = PackageManager()
+         pm.init(self.dir, False)
+         pkg = pm[name]
+         return pkg
+    
+    def compareInterface(self,interfaces):
+        test= True
+        n = 1
+    
+        while(test and n!=len(interfaces)):
+            if interfaces[n].min != interfaces[n-1].min: 
+                test=False
+                break
+            elif interfaces[n].max!=interfaces[n-1].max: 
+                test=False
+                break        
+            elif interfaces[n].step!=interfaces[n-1].step: 
+                test=False
+                break
+            elif type(interfaces[n])!=type(interfaces[n-1]): 
+                test=False
+                break
+            n=n+1
+        return test
+    
+
+    def compositeNodeInputs(self):
+        ins=[]
+        names = []
+   
+        for link in self.inputLinks:
+            name = link["source"]
+            if name not in names:
+                names.append(name)
+        # some models can have the same inputs. These inputs must have the same interfaces        
```

Comment:
I don't understand. If you use an OrderedDict, you will have all the inputs oredered, and no duplication.